### PR TITLE
Antora-based documentation build

### PR DESCRIPTION
**Pull request by zregvart**
This adds Antora-based documentation build. This relies on YAML files
generated by `internal/documentation` and an Antora extension in
`docs/extension` to generate content from those YAML files at runtime
using Handlebars template in `docs/reference.hbs`.

### CODE REVIEW

**Comment by zregvart** [comment]
Preview is [here](https://zregvart.github.io/ec-cli/)

**Comment by simonbaird** [comment]
I'm trying to get my head around it. Did you intend to check in the `docs/content/docs/*.md` here? If so, why?

**Comment by zregvart** [comment]
> I'm trying to get my head around it. Did you intend to check in the `docs/content/docs/*.md` here? If so, why?

Not at all, though I was certain did `git rm` them, removed now.

**Comment by simonbaird** [comment]
Checking if I understand why the docs/reference/*.yaml files are checked in: With those files checked in it means another antora build could import this repo and it would find the extension and everything would work. Is that right?

**Comment by zregvart** [comment]
> Checking if I understand why the docs/reference/*.yaml files are checked in: With those files checked in it means another antora build could import this repo and it would find the extension and everything would work. Is that right?

To a degree, the distribution of the extension is something that we need to do as well. Currently it is sourced from the repository itself and that works for this repository. We'd either need to push `reference.js` to NPM for Antora to be able to load it from another repository (and have it declared in that repositories `package.json`); or copy of it at build time so it can be [resolved](https://gitlab.com/antora/user-require-helper#user-content-usage).

**Comment by zregvart** [comment]
With [Publish release-antora-extension as GitHub package](https://github.com/hacbs-contract/ec-cli/pull/136/commits/39fa4b210074e12254c58acc355927cefb970e85) the Antora Extension that converts Cobra YAML to Asciidoc (`reference-antora-extension`) is published to GitHub Packages NPM repository which makes it available to other repositories.

One note is that GitHub personal access token with `read:packages` scope is required to fetch from GitHub's NPM repository see the [documentation](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#authenticating-with-a-personal-access-token) on how to do that.

I've tested this against my fork and the package [there](https://github.com/zregvart/ec-cli/packages/1581371). I'm expecting that the first build of the website would fail as the release of `reference-antora-extension` hasn't yet been published to this GitHub organization. Also the `package-lock.json` will change once we can pull from this GitHub organization.

To test locally pull in [this commit](https://github.com/zregvart/ec-cli/commit/374cddcb9ae16d405ab2f55a881368fc07c86993) over the branch here.

Latest preview is [here](https://zregvart.github.io/ec-cli/ec-cli/main/index.html)

File: reference-antora-extension/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            content.files.push(page)
+            referenceDocs.push({
+                path: basename,
+                name: data.name,
+            })
+        })
+
+        const nav = content.files.find(f => f.path.endsWith('nav.adoc'))
+        const navTemplate = Handlebars.compile(nav.contents.toString());
+        nav.contents = Buffer.from(navTemplate({ reference: referenceDocs }))
```
Comment (simonbaird) [inline reference-antora-extension/index.js:49]: I just realized how this part works. Very clever!

**Comment by bartoszmajsak** [comment]
Cool stuff indeed! I wrote an extension back in the day to invoke Cobra command help (in custom asciidoc format) from within Antora itself while building the docs (e.g. `include::cmd:ike[args='develop --help --help-format=adoc']` which becomes `https://istio-workspace-docs.netlify.app/istio-workspace/latest/cli_reference.html#ike-develop`). I might go back to it and revise it a bit based on your solution :) Thanks for sharing.

**Review by simonbaird — APPROVED** [review]
Lgtm.

**Pull request merged**